sherpa/astro/data.py

```diff
@@ -1305,31 +1305,54 @@ class DataPHA(Data1D):
 
     Notes
     -----
-    The original data is stored in the attributes - e.g. `counts` - and
-    the data-access methods, such as `get_dep` and `get_staterror`,
-    provide any necessary data manipulation to handle cases such as:
-    background subtraction, filtering, and grouping.
+    The original data is stored in the attributes - e.g. `counts` -
+    and the data-access methods, such as `get_dep` and
+    `get_staterror`, provide any necessary data manipulation to handle
+    cases such as: background subtraction, filtering, and grouping.
+
+    There is additional complexity compared to the Data1D case when
+    filtering data because:
+
+    * although the data uses channel numbers, users will often want to
+      filter the data using derived values (in energy or wavelength
+      units, such as 0.5 to 7.0 keV or 16 to 18 Angstroms);
+
+    * although derived from the Data1D case, PHA data is more-properly
+      thought about as being an integrated data set, so each channel
+      maps to a range of energy or wavelength values;
+
+    * the data is often grouped to improve the signal-to-noise, and so
+      requests for values need to determine whether to filter the data
+      or not, whether to group the data or not, and how to combine the
+      data within each group;
+
+    * and there is also the quality array, which indicates whether or
+      not a channel is trust-worthy or not (and so acts as an
+      additional filtering term).
 
     The handling of the AREASCAl value - whether it is a scalar or
-    array - is currently in flux. It is a value that is stored with the
-    PHA file, and the OGIP PHA standard ([1]_) describes the observed
-    counts being divided by the area scaling before comparison to the
-    model. However, this is not valid for Poisson-based statistics, and
-    is also not how XSPEC handles AREASCAL ([2]_); the AREASCAL values
-    are used to scale the exposure times instead. The aim is to add
-    this logic to the instrument models in `sherpa.astro.instrument`,
-    such as `sherpa.astro.instrument.RMFModelPHA`. The area scaling still
-    has to be applied when calculating the background contribution to
-    a spectrum, as well as when calculating the data and model values used
-    for plots (following XSPEC so as to avoid sharp discontinuities where
-    the area-scaling factor changes strongly).
+    array - is currently in flux. It is a value that is stored with
+    the PHA file, and the OGIP PHA standard ([1]_, [2]_) describes the
```

Comment:
The thing is, I do not see a warning from sphinx about these, and the links work - see https://sherpa--1264.org.readthedocs.build/en/1264/data/api/sherpa.astro.data.DataPHA.html#sherpa.astro.data.DataPHA

This is in the sherpa/astro/data.py file and we see no warnings (this is without the change in #1268 for the sphinx role, no idea what difference that makes).

```
% python setup.py build_sphinx >& loggy
% grep astro/data loggy
%
```

The problem lines look like

```
% grep reference loggy | head -3
reading sources... [ 41%] overview/api/sherpa.utils.guess_reference
/home/dburke/sherpa/sherpa-master/sherpa/astro/ui/utils.py:docstring of sherpa.astro.ui.utils.Session.calc_kcorr:3: WARNING: Duplicate target name, cannot be used as a unique reference: "1".
/home/dburke/sherpa/sherpa-master/sherpa/astro/ui/utils.py:docstring of sherpa.astro.ui.utils.Session.calc_kcorr:3: WARNING: Duplicate target name, cannot be used as a unique reference: "2".
```

so we can see what files cause the problems:

```
% grep reference loggy | sed -e 's/:.*//' | sort | uniq -c
     24 /home/dburke/sherpa/sherpa-master/sherpa/astro/ui/utils.py
     58 /home/dburke/sherpa/sherpa-master/sherpa/ui/utils.py
      1 reading sources... [ 41%] overview/api/sherpa.utils.guess_reference
      1 writing output... [ 41%] overview/api/sherpa.utils.guess_reference
```

So, it only the utils modules. I believe this comes from the fact we duplicate the functions from the Session object into a top-level function. To test this I took the first case (`calc_kcorr`) and changed the first two references to be unique:

```diff
-        The K correction ([1]_, [2]_, [3]_, [4]_) is the numeric
+        The K correction ([K1]_, [K2]_, [3]_, [4]_) is the numeric
```

With the original run we got

```
% grep calc_kcorr loggy
reading sources... [ 53%] ui/api/sherpa.astro.ui.calc_kcorr
/home/dburke/sherpa/sherpa-master/sherpa/astro/ui/utils.py:docstring of sherpa.astro.ui.utils.Session.calc_kcorr:3: WARNING: Duplicate target name, cannot be used as a unique reference: "1".
/home/dburke/sherpa/sherpa-master/sherpa/astro/ui/utils.py:docstring of sherpa.astro.ui.utils.Session.calc_kcorr:3: WARNING: Duplicate target name, cannot be used as a unique reference: "2".
/home/dburke/sherpa/sherpa-master/sherpa/astro/ui/utils.py:docstring of sherpa.astro.ui.utils.Session.calc_kcorr:3: WARNING: Duplicate target name, cannot be used as a unique reference: "3".
/home/dburke/sherpa/sherpa-master/sherpa/astro/ui/utils.py:docstring of sherpa.astro.ui.utils.Session.calc_kcorr:3: WARNING: Duplicate target name, cannot be used as a unique reference: "4".
writing output... [ 53%] ui/api/sherpa.astro.ui.calc_kcorr
```

with the new run we get the same

```
% python setup.py build_sphinx >& loggy2
% grep calc_kcorr loggy2
reading sources... [ 53%] ui/api/sherpa.astro.ui.calc_kcorr
/home/dburke/sherpa/sherpa-master/sherpa/astro/ui/utils.py:docstring of sherpa.astro.ui.utils.Session.calc_kcorr:66: WARNING:
duplicate citation K1, other instance in /home/dburke/sherpa/sherpa-master/docs/ui/api/sherpa.astro.ui.calc_kcorr.rst
/home/dburke/sherpa/sherpa-master/sherpa/astro/ui/utils.py:docstring of sherpa.astro.ui.utils.Session.calc_kcorr:69: WARNING:
duplicate citation K2, other instance in /home/dburke/sherpa/sherpa-master/docs/ui/api/sherpa.astro.ui.calc_kcorr.rst
/home/dburke/sherpa/sherpa-master/sherpa/astro/ui/utils.py:docstring of sherpa.astro.ui.utils.Session.calc_kcorr:3: WARNING: Duplicate target name, cannot be used as a unique reference: "3".
/home/dburke/sherpa/sherpa-master/sherpa/astro/ui/utils.py:docstring of sherpa.astro.ui.utils.Session.calc_kcorr:3: WARNING: Duplicate target name, cannot be used as a unique reference: "4".
writing output... [ 53%] ui/api/sherpa.astro.ui.calc_kcorr
```

So 

- I do not believe we have to change these references to be unique (as I don't think we have a page which includes multiple functions where the duplication would be a problem)
- I do not believe it would solve the problem

Reply:
OK I apparently need to look into the sphinx warnings in more detail later. I *thought* I understood the problem, but maybe not. Please leave it as it is here and I'll keep that on my internal list.